src/apps/hw.js

```diff
@@ -31,6 +35,147 @@ export const execWithTransport = (transport: Transport<*>): Exec => (
 
 const appsThatKeepChangingHashes = ["Fido U2F"];
 
```

Comment:
Ok, I don't fully understand why we have to do so much code in this file. I feel it's getting repetitive with code in here.

specifically, why wasn't it possible to just use `listApps` function & then we pipe in this => https://github.com/LedgerHQ/ledger-live-common/blob/master/src/apps/runner.js#L61-L69
?

Not only we write again the same logic but also I feel this logic here don't cover all cases. typically, Bitcoin app might be outdated and we need to uninstall and reinstall it all. we have to use the same dependency logic because it's going to evolve in future and we don't want to create two legacy to maintain.



On paper, i think it could work like this:

```js
export const standaloneAppOp = (
  transport: Transport<*>,
  appOp: AppOp,
  exec: Exec
): Observable<State> =>
  concat(
    of({ type: "listing-apps" }),
    from(getDeviceInfo(transport)).concat(
      mergeMap((deviceInfo) =>
        listApps(transport, deviceInfo).pipe(
          filter((e) => e.type === "result"),
          map((e) => e.result),
          mergeMap((listAppsResult) =>
            runOneAppOp(initState(listAppsResult), appOp, exec)
          )
        )
      )
    )
  );

```

the only thing is it's not 1:1 with your types: (need Exec, also it returns observable of State, but it's relatively easy to map that to more precise information. specifically, you can have info of the apps to update, and progress events)